[10.0][MIG] sale

rruebner opened the pull request:
FYI: I also updated the analysis text file for the sale module here. The rest of the analysis text files were updated in #850.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

rruebner committed:
[10.0][MIG] sale

cubells commented on odoo/openupgrade/doc/source/modules90-100.rst:
```diff
@@ -429,7 +429,7 @@ missing in the new release are marked with |del|.
 |resource                           | Nothing to do                     |
 +-----------------------------------+-----------------------------------+
-|sale                               |                                   |
+|sale                               | Done                                  |
```
Typo: there are 4 extra spaces. :)

cubells reviewed (changes requested):
Typo

pedrobaeza commented on addons/sale/migrations/10.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -14,7 +23,6 @@ sale         / sale.layout_category     / subtotal (boolean)            : NEW
 # NOTHING TO DO
```
This is not correct, as sale now integrates sale_layout module. You need to run update_module_names with the merge option activated and check if there's any change on data model

pedrobaeza commented on addons/sale/migrations/10.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -1,10 +1,19 @@
+sale         / product.template         / sale_line_warn_msg (text)     : NEW 
+sale         / res.partner              / sale_warn (selection)         : NEW required: required, selection_keys: ['block', 'no-message', 'warning'], req_default: no-message
+sale         / res.partner              / sale_warn_msg (text)          : NEW 
+# NOTHING TO DO
```
Please check it if warning module was installed on v9 and rescue values from it.

pedrobaeza commented on addons/sale/migrations/10.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,14 @@
+@openupgrade.migrate(use_env=False)
+def migrate(cr, version):
+    # map old / non existing value 'cost' to default value 'order'
+    cr.execute("""
```
You can use openupgradelib map_values function for readibility and extra log.

rruebner commented:
@pedrobaeza I updated the migration parts according to your mentioned points. In addition I rechecked all parts again and added some more migration parts.

pedrobaeza commented on addons/sale/migrations/10.0.1.0/pre-migration.py:
```diff
@@ -0,0 +1,132 @@
+    fields sale_line_warn, sale_line_warn_msg, sale_warn and sale_warn_msg are
+    moved from the old warning module to the sale module. Other fields are
+    moved in other modules (e. g. field purchase_warn in purchase module). If
+    we would use openupgrade.update_module_names their might be problems when
```
s/their/there

pedrobaeza commented on addons/sale/migrations/10.0.1.0/pre-migration.py:
```diff
@@ -0,0 +1,132 @@
+    moved from the old warning module to the sale module. Other fields are
+    moved in other modules (e. g. field purchase_warn in purchase module). If
+    we would use openupgrade.update_module_names their might be problems when
+    first use openupgrade.update_module_names in sale module and than again in
```
s/use/using

pedrobaeza replied:
s/than/then

pedrobaeza reviewed (changes requested):
Check if you need to rename the security groups

rruebner commented:
@pedrobaeza I fixed the mentioned typos and rechecked security group changes. AFAICS there is no extra renaming needed (possible renaming is covered by update_module_names calls).
I also checked for noupdate=1 records and everything was fine.

MiquelRForgeFlow reviewed (approved):
LGTM! :+1:

pedrobaeza commented:
Thanks for the changes, Robert.

If you don't mind, I'm going to keep the PRs unmerged until we get a green branch for not adding more complexity to the pipe and possible side effects.

MiquelRForgeFlow commented:
Please, rebase the PR (and squash minor commits) ☺️

MiquelRForgeFlow reviewed (changes requested):
A new line appeared https://github.com/OCA/OpenUpgrade/pull/917/files#diff-1c07737aeb217d5e147f8fa6d1204414R5, and it should be attended in an easy post-migration script.

omar7r commented on addons/sale/migrations/10.0.1.0/pre-migration.py:
```diff
@@ -0,0 +1,132 @@
+
+# backup of separator column in table sale_layout_category because it doesn't
+# exist anymore in odoo 10
+column_renames = {
```
If you don't have installed "sale_layout" module, columns renaming and the models renaming fails, maybe, you have to check if this module is installed to execute this two functions

pedrobaeza committed:
[FIX] sale: Don't fail w/o sale_layout

pedrobaeza commented:
This fails due to TRAVIS_COMMIT, but I have checked it and fix it, so I merge.
